enhancements/machine-config/user-data-secret-managed.md

```diff
@@ -0,0 +1,113 @@
+
+## Proposal
+
+The proposal is pretty simple today:
+
+- The installer keeps creating the Secret that holds the stub ignition config (at any given version) under a new name
+- The secret will change name from `<role>-user-data` to `<role>-user-data-managed` which is going to be v3 at a later time before the release
+- The name change is needed to keep supporting old bootimages that still reference the old ignition version
+- The MCO takes over owning that Secret and its content once the cluster is up and running
+- The MCO has a way to manage the lifecycle of the Secret containing the stub ignition
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+The implementation has already started and there are two PRs that provide a working POC:
+
+- https://github.com/openshift/machine-config-operator/pull/1792
+- https://github.com/openshift/installer/pull/3730
+
+The MCO PR simply learns how to create the content of the secret and once it's running, it goes ahead and takes ownership of the secret in the cluster (you can see we're using the new name ending in `-managed`). That allows us to upgrade the MCO to a newer Ignition version and make sure the secret's content is updated as well.
+The installer PR is just a rename of the old user-data secret to something new that the MCO can manage. The main reason to the rename is ability to still have the old user-data secret around during an upgrade.
+
+The whole implementation, related to the ignition upgrade will go as follows:
+
+- installation (no impact)
+    - everything starts with the new ignition version
+    - installer creates the user-data with ignition v3
+    - MCO is ignition v3 as well
+    - scale up works as the new bootimages understand ignition v3
+
+- upgrade (impacted)
+    - everything is ignition v2, including bootimages
+    - the user-data secret is also an ignition v2 snippet
+    - the upgrade starts bringing the new MCO which understands ignition v3
+    - the MCO creates the new managed secret containing an ignition v3 snippet and leaving around the old, unmanaged secret which is v2
+    - scale up continues working as the Machine(Sets) still reference the unmanaged, v2 snippet
+    - when we'll grow the ability to update bootimages, the new bootimages will understand v3 and they'll grab the managed secret which is v3
+
+
+### Risks and Mitigations
+
+This enhancement isn't about the migration to Ignition v3. Any risks and mitigations related to that won't be captured here. This is going to be a straightforward implementation during the development cycle leading to the final migration (which will obviously require more coordinations and/or ratcheting strategy). The only thing to note related to this enhacement is the need to coordinate 2 code changes in both MCO and installer, with the MCO one needed first.
+
+
+## Drawbacks
+
```

Comment:
Another drawback is this make existing upgraded clusters look different from brand new clusters >=4.6.

Reply:
and that is because machinesets are still at the old secret and we have two secrets for the old and new ign version right?

Reply:
also, we kind of knew that already, as upgrading ignition itself is a big change per se - we're trying to minimize changes (and we all would have had bootimages upgrade before this..)